### PR TITLE
Bohdan

### PR DESCRIPTION
**Pull request by BogdanKalysh**

### CODE REVIEW

File: budget_manager_backend/sources/handlers/categoryhandler.cpp [hunk 1]
```diff
@@ -73,10 +73,14 @@ void CategoryHandler::post(Poco::Net::HTTPServerRequest& request, Poco::Net::HTT
 }
 
 void CategoryHandler::del(Poco::Net::HTTPServerRequest& request, Poco::Net::HTTPServerResponse& response){
-    QJsonObject bodyObj = convertIstreamToJson(request.stream());
-    Category category = parser->parse(bodyObj);
+//    QJsonObject bodyObj = convertIstreamToJson(request.stream());
```
Comment (AndriyFedorenko) [inline budget_manager_backend/sources/handlers/categoryhandler.cpp:76]: It is not a good practice leave commented code. You can use git history for this

File: budget_manager_frontend/sources/listItems/TransactionsListItem.cpp [hunk 2]
```diff
@@ -20,10 +27,64 @@ TransactionsItem::TransactionsItem(Transaction transaction, QWidget *parent) : Q
...
+}
+
+bool TransactionsItem::event(QEvent * e)
+{
+    switch(e->type())
+    {
+    case QEvent::HoverEnter:
+        hoverEnter(static_cast<QHoverEvent*>(e));
+        return true;
+        break;
```
Comment (AndriyFedorenko) [inline budget_manager_frontend/sources/listItems/TransactionsListItem.cpp:66]: You don't need break if you use return

File: budget_manager_frontend/sources/windows/mainWindow.cpp [hunk 3]
```diff
@@ -172,12 +169,19 @@ void MainWindow::configWindowItems()
 }
 
 
-qreal MainWindow::getCategoryTotalSum(QString categoryName)
+qreal MainWindow::getCategoryTotalSum(QString categoryName, Type type)
 {
+    QString typestr;
```
Comment (AndriyFedorenko) [inline budget_manager_frontend/sources/windows/mainWindow.cpp:174]: Simplify

QString typestr = type ? jsonbuilder::INCOME : jsonbuilder::EXPENSE;

File: budget_manager_frontend/sources/windows/usersettingsdialog.cpp [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        ui->categoritesComboBox->addItem(cat.getName(), cat.getId());
+    }
+
+    categoriesReply->close();
+    categoriesReply->deleteLater();
+}
+
+void UserSettingsDialog::on_saveButton_clicked()
+{
+
```
Comment (AndriyFedorenko) [inline budget_manager_frontend/sources/windows/usersettingsdialog.cpp:69]: remove empty line

File: budget_manager_frontend/sources/windows/usersettingsdialog.cpp [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    if(newuser.getName() == "")
+    {
+        QMessageBox::about(this, "info", "Введіть ім'я");
+    }
+    else if(!emailRegExp.match(newuser.getEmail()).hasMatch())
+    {
+        QMessageBox::about(this, "info", "Введено некоректний Email");
+    }
+    else
```
Comment (AndriyFedorenko) [inline budget_manager_frontend/sources/windows/usersettingsdialog.cpp:129]: You use different code styles

File: budget_manager_frontend/sources/windows/usersettingsdialog.cpp [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+    getUserReply->close();
+    getUserReply->deleteLater();
+}
+
+void UserSettingsDialog::updateUser(User& newuser)
+{
+    QRegularExpression emailRegExp("^[a-z0-9]([a-z0-9.]+[a-z0-9])?\\@[a-z0-9.-]+$");
+
+    if(newuser.getName() == "")
```
Comment (AndriyFedorenko) [inline budget_manager_frontend/sources/windows/usersettingsdialog.cpp:121]: string should have method empty. Please use this method

File: budget_manager_frontend/sources/windows/usersettingsdialog.cpp [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+        QMessageBox::about(this, "info", "Введено некоректний Email");
+    }
+    else
+    {
+        QScopedPointer<IJsonBuilder<User>> builder (new UserJsonBuilder);
+        QJsonObject jsonData =  builder->buildJson(newuser);
+        QJsonDocument jsonDoc(jsonData);
+        QByteArray byteData = jsonDoc.toJson();
+        QNetworkRequest request = QNetworkRequest(QUrl(jsonbuilder::USERURL));
+        request.setRawHeader("Content-Type", "application/json");
```
Comment (AndriyFedorenko) [inline budget_manager_frontend/sources/windows/usersettingsdialog.cpp:136]: move it to constants

File: budget_manager_frontend/sources/windows/usersettingsdialog.cpp [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+}
+
+void UserSettingsDialog::finishedUpdateUser()
+{
+    QNetworkReply *getUserReply = qobject_cast<QNetworkReply*>(sender());
+
+    if(getUserReply->error() == QNetworkReply::NoError){
+        user.setEmail(ui->emailInputField->text());
+        user.setName(ui->nameInputField->text());
+        if(ui->newPasswordInput->text() != ""){
```
Comment (AndriyFedorenko) [inline budget_manager_frontend/sources/windows/usersettingsdialog.cpp:150]: Please use empty

File: budget_manager_frontend/sources/windows/usersettingsdialog.cpp [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+    if(contents != ""){
+        QRegularExpression passwordRegExp("^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d]{8,}$");
+
+        QString newName = ui->nameInputField->text();
+        QString newEmail = ui->emailInputField->text();
+        QString newPassword = QCryptographicHash::hash(ui->newPasswordInput->text().toLocal8Bit(), QCryptographicHash::Sha224).toHex().data();
+
+        if(!passwordRegExp.match(ui->newPasswordInput->text()).hasMatch())
+        {
+            QMessageBox::about(this, "info", "Введено некоректний новий пароль.");
```
Comment (AndriyFedorenko) [inline budget_manager_frontend/sources/windows/usersettingsdialog.cpp:102]: I recommend move all this UI text to one file and define some constants

**Review by AndriyFedorenko — COMMENTED** [review]
Please write code in one style

File: budget_manager_backend/sources/handlers/categoryhandler.cpp (hunk 1 above)
Comment (BogdanKalysh) [reply budget_manager_backend/sources/handlers/categoryhandler.cpp:76]: removed comments

File: budget_manager_frontend/sources/listItems/TransactionsListItem.cpp (hunk 2 above)
Comment (BogdanKalysh) [reply budget_manager_frontend/sources/listItems/TransactionsListItem.cpp:67]: removed breaks

File: budget_manager_frontend/sources/windows/mainWindow.cpp (hunk 3 above)
Comment (BogdanKalysh) [reply budget_manager_frontend/sources/windows/mainWindow.cpp:177]: used ternary operator

File: budget_manager_frontend/sources/windows/usersettingsdialog.cpp (hunk 6 above)
Comment (BogdanKalysh) [reply budget_manager_frontend/sources/windows/usersettingsdialog.cpp:128]: used isEmpty()

File: budget_manager_frontend/sources/windows/usersettingsdialog.cpp (hunk 7 above)
Comment (BogdanKalysh) [reply budget_manager_frontend/sources/windows/usersettingsdialog.cpp:143]: added constants

File: budget_manager_frontend/sources/windows/usersettingsdialog.cpp (hunk 8 above)
Comment (BogdanKalysh) [reply budget_manager_frontend/sources/windows/usersettingsdialog.cpp:158]: Used isEmpty()

File: budget_manager_frontend/sources/windows/usersettingsdialog.cpp (hunk 9 above)
Comment (BogdanKalysh) [reply budget_manager_frontend/sources/windows/usersettingsdialog.cpp:107]: Created constants

File: budget_manager_frontend/sources/windows/usersettingsdialog.cpp (hunk 4 above)
Comment (BogdanKalysh) [reply budget_manager_frontend/sources/windows/usersettingsdialog.cpp:74]: removed

File: budget_manager_frontend/sources/windows/usersettingsdialog.cpp (hunk 5 above)
Comment (BogdanKalysh) [reply budget_manager_frontend/sources/windows/usersettingsdialog.cpp:136]: will refactor code

**Pull request merged**